core/offchain/offchain_worker.hpp

```diff
@@ -32,7 +32,7 @@ namespace kagome::offchain {
    public:
     virtual ~OffchainWorker() = default;
 
-    virtual outcome::result<void> run() = 0;
+    virtual void run(std::function<void()> &&func, std::string label) = 0;
```

Comment:
```suggestion
    virtual void run(const BlockHeader &) = 0;
```

Reply:
No, this is delegated to calling code

Reply:
Sassafras will be publish tickets by off-chain workers

Reply:
It requires "offchain context", not "offchain thread"